### ISSUE TITLE
SystemError when running tests with numpy-1.11 b2

### ISSUE THREAD

**Issue by olebole** [spacetelescope/pyasdf#189]
When running with the 1.11 b2 beta release of numpy (Debian unstable), I get a couple of errors like
```
____________________________ test_streams[tree0] ______________________________

tree = {'not_shared': array([10,  9,  8,  7,  6,  5,  4,  3,  2,  1], dtype=uint8),
     'science_data': array([ 0.,  1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  9.]), 'skipping

    def test_streams(tree):
        buff = io.BytesIO()
    
        def get_write_fd():
            return generic_io.OutputStream(buff)
    
        def get_read_fd():
            buff.seek(0)
            return generic_io.InputStream(buff, 'rw')
    
>       with _roundtrip(tree, get_write_fd, get_read_fd) as ff:

pyasdf/tests/test_generic_io.py:226: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pyasdf/tests/test_generic_io.py:59: in _roundtrip
    ff = asdf.AsdfFile.open(fd, **read_options)
pyasdf/asdf.py:533: in open
    do_not_fill_defaults=do_not_fill_defaults)
pyasdf/asdf.py:475: in _open_impl
    fd, past_magic=True, validate_checksums=validate_checksums)
pyasdf/block.py:243: in read_internal_blocks
    block = self._read_next_internal_block(fd, past_magic=past_magic)
pyasdf/block.py:211: in _read_next_internal_block
    validate_checksum=self._validate_checksums)
pyasdf/block.py:989: in read
    fd, self._size, self._data_size, self.compression)
pyasdf/block.py:1002: in _read_data
    return fd.read_into_array(used_size)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pyasdf.generic_io.InputStream object at 0x7f4b413b5710>, size = 80

    def read_into_array(self, size):
        try:
            # See if Numpy can handle this as a real file first...
>           return np.fromfile(self._fd, np.uint8, size)
E           SystemError: error return without exception set

pyasdf/generic_io.py:870: SystemError
```
This happens for `teststreams`, `testurlopen`, `testhttpconnection`, `testexplodedhttp`, `testseekuntilonblockboundary`, `teststreamtostream`, and `testarraytostream`.
With numpy-1.10 (Debian testing) the tests run fine.

I have no idea whether this is an issue for numpy or for pyasdf; I will forward is to numpy as well.

**Comment by njsmith** [comment]
Definitely a bug in numpy.

**Comment by olebole** [comment]
Would you then subscribe to numpy/numpy#7200 please, in case that a context is needed to discuss the bug?

**Comment by njsmith** [comment]
@olebole: I'm subscribed to all bugs on the numpy tracker, that's how I saw this :-)

**Comment by olebole** [comment]
Since this is a numpy problem (and there it is solved, hopefully delivered with 1.11.0), I am closing it here now.
